lib/queue_classic/worker.rb

```diff
@@ -161,5 +163,15 @@ def setup_queues(adapter, queue, queues, top_bound)
       end
     end
 
+    def clone_args(job)
+      job[:args].map do |arg|
+        begin
+          arg.clone
+        rescue TypeError
```

Comment:
Oh, thanks for the explanation!

I would love to see an explicit test for this unless it's hard to do, but it shouldn't from what I understand.